### PR TITLE
[docs] Generate API docs.

### LINKED ISSUE #1249: Add pages with automatically generated API documentation
**What would you like to be added**:

API documentation auto-generated from the types.go files.

/kind documentation

**Why is this needed**:

This is an easy way to browse the definition of every field of the API.

We should investigate how kubernetes API reference is generated and use similar tools

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [ ] API change
- [ ] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Generate and publish API reference documentation for Kueue.

#### Which issue(s) this PR fixes:

Fixes #1249 

#### Special notes for your reviewer:

There is a PR
* kubernetes-sigs/reference-docs#343 

that will improve the output for map types.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: site/static/ref/kueue.v1beta1.md
Comment (alculquicondor) [inline site/static/ref/kueue.v1beta1.md:1]: Is this an old location? Do we need both this and `content/en/docs/reference`?

**Review by alculquicondor — COMMENTED** [review]
It looks like the site generation failed, but it might be a fluke. Maybe amend the commit?

File: site/genref/config.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+hiddenMemberFields:
+  - "TypeMeta"
+  - "ObjectMeta"
+
+apis:
+  - name: kueue
+    title: Kueue API
+    package: ../..
+    path: apis/kueue/v1beta1
```
Comment (alculquicondor) [inline site/genref/config.yaml:9]: Is it possible to add a description that can be shown in the reference page?

https://deploy-preview-1263--kubernetes-sigs-kueue.netlify.app/docs/reference/

Comment (trasc) [reply site/genref/config.yaml:9]: Yes, from the templates. Done.

File: Makefile [hunk 2]
```diff
@@ -177,8 +177,8 @@ ci-lint: golangci-lint
 	$(GOLANGCI_LINT) run --timeout 15m0s
 
 .PHONY: verify
-verify: gomod-verify vet ci-lint fmt-verify toc-verify manifests generate update-helm
-	git --no-pager diff --exit-code config/components apis charts/kueue/templates client-go
+verify: gomod-verify vet ci-lint fmt-verify toc-verify manifests generate update-helm generate-apiref
+	git --no-pager diff --exit-code config/components apis charts/kueue/templates client-go site/content/en/docs/reference/kueue.v1beta1.md
```
[the remark this answers is not in the document]
Comment (trasc) [reply Makefile]: done

File: Makefile [hunk 3]
```diff
@@ -314,6 +314,11 @@ HELM = $(PROJECT_DIR)/bin/helm
 helm: ## Download helm locally if necessary.
 	@GOBIN=$(PROJECT_DIR)/bin GO111MODULE=on $(GO_CMD) install helm.sh/helm/v3/cmd/helm@v3.12.1
 
+GENREF = $(PROJECT_DIR)/bin/genref
+.PHONY: genref
+genref: ## Download genref locally if necessary.
+	@GOBIN=$(PROJECT_DIR)/bin GO111MODULE=on $(GO_CMD) install github.com/kubernetes-sigs/reference-docs/genref@latest
```
[the remark this answers is not in the document]
Comment (trasc) [reply Makefile]: pinned to the current master, I'll ask about releases but I don't think it will happen.

File: site/static/ref/kueue.v1beta1.md
Comment (trasc) [reply site/static/ref/kueue.v1beta1.md:1]: This is an old one, removed.

**Pull request merged**